crawl-ref/source/dgn-overview.cc

```diff
@@ -1093,3 +1062,164 @@ bool connected_branch_can_exist(branch_type br)
 
     return true;
 }
+
+/**
+ * Make the little overview
+ * (D) Dungeon        (T) Temple         (L) Lair           etc.
+ * at most 4 branches on 1 line
+*/
+static void _show_dungeon_overview(vector<branch_type> brs)
+{
+    clear_messages();
+    int linec = 0;
+    string line;
+    for (branch_type br : brs)
+    {
+        if (linec == 4)
+        {
+            linec = 0;
+            mpr(line);
+            line = "";
+        }
+        line += make_stringf("(%c) %-14s ",
+                             branches[br].travel_shortcut,
+                             branches[br].shortname);
+        ++linec;
+    }
+    if (!line.empty())
+        mpr(line);
+    return;
```

Comment:
I think you'll need some kind of refresh call here to get this to show. I haven't tested so I'm not sure exactly which is needed but maybe `flush_prev_message();`.